Types/Payload.php

```diff
@@ -19,30 +19,32 @@
 namespace Circle\DoctrineRestDriver\Types;
 
 use Circle\DoctrineRestDriver\Enums\SqlOperations;
-use Circle\DoctrineRestDriver\Validation\Assertions;
 
 /**
- * Payload type: Union type for UpdatePayload and InsertPayload
+ * Payload type: Union type for InsertChangeSet and InsertChangeSet
  *
  * @author    Tobias Hauck <tobias@circle.ai>
  * @copyright 2015 TeeAge-Beatz UG
  */
 class Payload {
 
     /**
-     * Returns an InsertPayload::create or UpdatePayload::create
+     * Returns an InsertChangeSet::create or InsertChangeSet::create
      * result or null depending on the given tokens array
      *
      * @param  array       $tokens
+     * @param  array       $options
      * @return null|string
      *
      * @SuppressWarnings("PHPMD.StaticAccess")
      */
-    public static function create(array $tokens) {
-        Assertions::assertHashMap('tokens', $tokens);
+    public static function create(array $tokens, array $options) {
+        HashMap::assert($tokens, 'tokens');
 
+        $format    = Format::create($options);
         $operation = SqlOperation::create($tokens);
+
         if ($operation === SqlOperations::SELECT || $operation === SqlOperations::DELETE) return null;
-        return $operation === SqlOperations::INSERT ? InsertPayload::create($tokens) : UpdatePayload::create($tokens);
+        return $format->encode($operation === SqlOperations::INSERT ? InsertChangeSet::create($tokens) : UpdateChangeSet::create($tokens));
```

Comment:
wie auch schonma gesagt: wenn du mehr als zwei branches has, lass den ternären weg, das is sons unübersichtlich. ich würde das bei sowas explizit machen, das is immer klarer bei so dispatches,sprich:

```php
if($operation === SqlOperations::INSERT)  return $format->encode(InsertChangeSet::create($tokens));
if($operation === SqlOperations::Update)  return $format->encode(UpdateChangeSet::create($tokens));

return null;
```

Der vollständigkeit halber (aber glaube, dass das mit php nich geht: Du müsstes eigtl den konstruktor nur dispatchen, also etwa so (und das is glaub ich, was du hier wolltest)

```js
const dispatch(op) {
    if(op === Ops.Insert)   return InsertChangeSet;
    if(op === Ops.Update) return UpdateChangeSet;
    
    // fake constructor, der immer null is
    return {
        create: () => null
    };
}

Payload.create = function(tokens) {
    const constr = dispatch(SqlOperation.create(tokens));
    const format = Format.create(options);

    // hierbei format.encode so überladen, dass es bei null einfach null returned,
    // damit das auch mit delete un so klappt
    format.encode(constr.create(tokens));
}
```

-> is übrigens en schönes beispiel für: ich pass mich der umwelt an vs. ich pass die umwelt mir an :D